source/extensions/filters/udp/dns_filter/dns_filter.cc

```diff
@@ -0,0 +1,57 @@
+#include "extensions/filters/udp/dns_filter/dns_filter.h"
+
+#include "envoy/network/listener.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace DnsFilter {
+
+DnsFilterEnvoyConfig::DnsFilterEnvoyConfig(
+    Server::Configuration::ListenerFactoryContext& context,
+    const envoy::config::filter::udp::dns_filter::v2alpha::DnsFilterConfig& config)
+    : root_scope(context.scope()), stats_(generateStats(config.stat_prefix(), root_scope)) {
+
+  using envoy::config::filter::udp::dns_filter::v2alpha::DnsFilterConfig;
+
+  // store configured data for server context
+  const auto& server_config = config.server_config();
+
+  if (server_config.has_inline_dns_table()) {
+
+    const auto& cfg = server_config.inline_dns_table();
+    const size_t entries = cfg.virtual_domains().size();
+
+    // TODO (abaptiste): Check that the domain configured here appears
+    // in the known domains list
+    virtual_domains_.reserve(entries);
+    for (const auto virtual_domain : cfg.virtual_domains()) {
+      DnsAddressList addresses{};
+
+      if (virtual_domain.endpoint().has_address_list()) {
+        const auto& address_list = virtual_domain.endpoint().address_list().address();
+        addresses.reserve(address_list.size());
+        for (const auto& configured_address : address_list) {
+          addresses.push_back(configured_address);
+        }
+      }
+
+      virtual_domains_.emplace(std::make_pair(virtual_domain.name(), addresses));
```

Comment:
Shouldn't this be
`virtual_domains_.emplace(virtual_domain.name(), addresses);` or even more efficiently `virtual_domains_.emplace(virtual_domain.name(), std::move(addresses));`